setup/symfony_server.rst

```diff
@@ -0,0 +1,271 @@
+
+Setting up the Local Proxy
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Local domains are possible thanks to a local proxy provided by the Symfony
+server. First, start the proxy:
+
+.. code-block:: terminal
+
+    $ symfony proxy:start
+
+If this is the first time you run the proxy, you must follow these additional steps:
+
+* Open the **network configuration** of your operating system;
+* Find the **proxy settings** and select the **"Automatic Proxy Configuration"**;
+* Set the following URL as its value: ``https://127.0.0.1:7080/proxy.pac``
+
+Defining the Local Domain
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, Symfony proposes ``.wip`` (for *Work in Progress*) as the local
+domains (but you can choose any other domain and TLD you like). Define a local
+domain for a project as follows:
+
+.. code-block:: terminal
+
+    $ cd my-project/
+    $ symfony proxy:domain:attach my-domain.wip
+
+If you have installed the local proxy as explained in the previous section, you
+can now browse ``https://my-domain.wip`` to access to your local project with
+the new custom domain.
+
+.. tip::
+
+    Browse the https://127.0.0.1:7080 URL to get the full list of local project
+    directories, their custom domains, and port numbers.
+
+When running console commands, add the ``HTTPS_PROXY`` env var to make custom
+domains work:
+
+.. code-block:: terminal
+
+    $ HTTPS_PROXY=https://127.0.0.1:7080 curl https://my-domain.wip
+
+Long-Running Commands
+---------------------
+
+Long-running commands, such as the ones related to compiling front-end web
+assets, block the terminal and you can't run other commands. The Symfony server
+provides a ``run`` command to wrap them as follows:
+
+.. code-block:: terminal
+
+    # compile Webpack assets using Symfony Encore ... but do that in the
+    # background to not block the terminal
+    $ symfony run -d yarn encore dev --watch
+
+    # continue working and running other commands...
+
+    # from time to time, check the command logs if you want
+    $ symfony server:log
+
+    # and you can also check if the command is still running
+    $ symfony server:status
+    Web server listening on ...
+    Command "yarn ..." running with PID ...
+
+    # stop the command (and the whole server) when you are finished
+    $ symfony server:stop
+
+Bonus Features
+--------------
+
+The Symfony server is much more than a local web server and it includes other
+useful features.
+
+Looking for Security Vulnerabilities
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of installing the :doc:`Symfony Security Checker </security/security_checker>`
+as a dependency of your projects, you can use this command from the Symfony server:
+
+.. code-block:: terminal
+
+    $ symfony security:check
+
+This command uses the same vulnerability database as the Symfony Security
+Checker but it also caches that information to keep checking security when
+it's not possible to access to that public database.
```

Comment:
You can also add that it does not use the official API endpoint as everything is done locally (except for cloning the vulnerability DB). Which is probably the best in a CI scenario (adding a tip about this would be great).